### ISSUE TITLE
本地启动程序崩溃后, 之前的标注文件 json 如何重新复用? 如何复现崩溃前的标注状态?

### ISSUE THREAD

**Issue by zzx58-027** [CVHub520/X-AnyLabeling#1216]
### Search before asking

- [x] I have searched the X-AnyLabeling [Docs](https://github.com/CVHub520/X-AnyLabeling/tree/main/docs) and [issues](https://github.com/CVHub520/X-AnyLabeling/issues) and found no similar questions.

### Question

<img width="1478" height="1228" alt="Image" src="https://github.com/user-attachments/assets/e2709f9c-67a8-4e08-a300-b71c4f42df39" />
我将 json 结果保存在图片文件夹外的地方统一管理了, 程序崩溃后, 如何将这些 json 和对应图片对应起来, 让程序可以使用?

**Comment by zzx58-027** [comment]
目前是将 json 结果文件放在图片同一路径下可以解决, 但是为什么要有将标注结果放到另外的文件夹下的支持?

**Comment by zzx58-027** [comment]
目前可以完全复现的情况, Mac M1 Pro, 打开 ultralytics 模型训练, data 选择框, 点开后弹出系统文件选择框, 不选择文件, 点击 cancel 就会导致程序崩溃.

**Comment by zzx58-027** [comment]
应该如何正确使用软件中附带的 ultralytics 训练模型?

**Comment by zzx58-027** [comment]
> 应该如何正确使用软件中附带的 ultralytics 训练模型?

https://github.com/CVHub520/X-AnyLabeling/blob/main/examples/training/ultralytics/README.md

**Comment by zzx58-027** [comment]
> 应该如何正确使用软件中附带的 ultralytics 训练模型?

文档说 data 栏不需要输入, 软件会为我们自动生成一个, 但是实际上必须提供 data.yaml? 我对自己如何书写 data.yaml 感到疑惑, 我的图片并没有划分 train 和 验证集, 我要如何做?

**Comment by zzx58-027** [comment]
<img width="2382" height="1654" alt="Image" src="https://github.com/user-attachments/assets/c8ca76a1-ba8e-4aa0-bd8a-0c26b111cdb6" />

<img width="2408" height="1654" alt="Image" src="https://github.com/user-attachments/assets/daf11c1b-6c58-4dbc-a6fe-73e7de133738" />

请问这个要如何解决?

**Comment by CVHub520** [comment]
Hi @zzx58-027, thanks for reaching out with these excellent questions and detailed crash report! We really appreciate your contributions to making X-AnyLabeling better. 

Regarding your question about reusing JSON files: X-AnyLabeling is designed to default to reading label information from the same directory as the image files, which often feels intuitive for many users. However, we also offer the flexibility to specify a custom label output directory if you prefer to manage your labels in a separate location, giving you more control over your project structure.

Thank you so much for the detailed steps to reproduce the crash! We've identified and fixed this specific issue (related to canceling the file selection dialog in Ultralytics training settings) in our latest `main` branch. Please update your source code, and that should resolve it for you. 🙏

About the Ultralytics training `data` field and `data.yaml`: Our documentation notes that while classification tasks don't require an uploaded `data.yaml`, other tasks indeed need one. For comprehensive guidance on structuring your `data.yaml`, I highly recommend consulting our official documentation as well as the Ultralytics official manual. You won't need to manually split your images into train/validation sets; X-AnyLabeling automatically handles this for your loaded task based on your chosen split ratio. Simply define your essential fields, like classes information, within the `data.yaml`, and the system will take care of the rest.
